endpoints/info/bidders.go

```diff
@@ -0,0 +1,92 @@
+package info
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+	"strings"
+
+	"github.com/golang/glog"
+	"github.com/julienschmidt/httprouter"
+	"github.com/prebid/prebid-server/openrtb_ext"
+	yaml "gopkg.in/yaml.v2"
+)
+
+// NewBiddersEndpoint implements /info/bidders
+func NewBiddersEndpoint() httprouter.Handle {
+	bidderNames := make([]string, 0, len(openrtb_ext.BidderMap))
+	for bidderName := range openrtb_ext.BidderMap {
+		bidderNames = append(bidderNames, bidderName)
+	}
+
+	jsonData, err := json.Marshal(bidderNames)
+	if err != nil {
+		glog.Fatalf("error creating /info/bidders endpoint response: %v", err)
+	}
+
+	return httprouter.Handle(func(w http.ResponseWriter, req *http.Request, _ httprouter.Params) {
+		w.Header().Set("Content-Type", "application/json")
+		if _, err := w.Write(jsonData); err != nil {
+			glog.Errorf("error writing response to /info/bidders: %v", err)
+		}
+	})
+}
+
+// NewBiddersEndpoint implements /info/bidders/*
+func NewBidderDetailsEndpoint(infoDir string) httprouter.Handle {
+	// Build all the responses up front, since there are a finite number and it won't use much memory.
+	files, err := ioutil.ReadDir(infoDir)
```

Comment:
Would it be better to use the same list of bidders generated above, and panic if the file for a bidder is missing? This assures that we will have info for all supported bidders, and not bother reading random files that somehow got into the directory.

Reply:
This is a good idea.... I'll change this shortly.

Reply:
Actually... one major problem with this is that our unit tests can't load the yaml files from the `sample` directory anymore. That makes the tests a bit less stable, since they're relying on prod code which could change at any time.

Maybe a better idea is to just update `TestInfoFiles` to make sure that that directory _only_ has files for known bidders in it?

Reply:
Probably best to fix up `TestInfoFiles`. Either add the production `yaml` file to the `sample` directory, or just have a default info file for all bidders you don't have specific tests for (you can then bulk test the rest if you want).

Reply:
oh... I misread this, and implemented something a bit different. I think this way is better, though.

I just added an assertion to check that `numFiles == len(bidderMap)`. If any extra files get added into the prod folder, that check will fail.

I like this better because it has a lower maintenance cost. We won't have to add a dummyfile to the test directory every time there's a new Bidder this way.

Reply:
Refactored some things as we discussed, trying to get the best balance of all the benefits.

The `BidderList` is now passed into the endpoint, so the tests can use the mock directory. The implementation now relies on the `BidderMap`, rather than the directory file contents, which should make it a bit more robust if people add or remove files between test time and deployment time.